## 1. Data 가져오기

In [421]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import os
import warnings
warnings.filterwarnings('ignore')


In [422]:
KP_2020 = pd.read_csv("C:/Users/user/Desktop/사본(73) - 활용데이터/KP2020.csv",encoding='cp949')
KP_2021 = pd.read_csv("C:/Users/user/Desktop/사본(73) - 활용데이터/KP2021.csv",encoding='cp949')
NPA_2020 = pd.read_csv("C:/Users/user/Desktop/사본(73) - 활용데이터/NPA2020.csv",encoding='cp949')
ASAN_CCTV = pd.read_csv("C:/Users/user/Desktop/충남,대전, 세종 무인교통단속카메라/asan.csv",encoding='cp949')
BORYEONG_CCTV = pd.read_csv("C:/Users/user/Desktop/충남,대전, 세종 무인교통단속카메라/boryeong.csv",encoding='cp949')
CHEONAN_CCTV = pd.read_csv("C:/Users/user/Desktop/충남,대전, 세종 무인교통단속카메라/cheonan.csv",encoding='cp949')
CHUNGYANG_CCTV = pd.read_csv("C:/Users/user/Desktop/충남,대전, 세종 무인교통단속카메라/chungyang.csv",encoding='cp949')
DAEJEON_CCTV = pd.read_csv("C:/Users/user/Desktop/충남,대전, 세종 무인교통단속카메라/daejeon.csv",encoding='cp949')
DANGJIN_CCTV = pd.read_csv("C:/Users/user/Desktop/충남,대전, 세종 무인교통단속카메라/dangjin.csv",encoding='cp949')
GEOMSAN_CCTV = pd.read_csv("C:/Users/user/Desktop/충남,대전, 세종 무인교통단속카메라/geomsan.csv",encoding='cp949')
SEJONG_CCTV = pd.read_csv("C:/Users/user/Desktop/충남,대전, 세종 무인교통단속카메라/sejong.csv",encoding='cp949')
SEOCHEON_CCTV = pd.read_csv("C:/Users/user/Desktop/충남,대전, 세종 무인교통단속카메라/seocheon.csv",encoding='cp949')
SEOSAN_CCTV = pd.read_csv("C:/Users/user/Desktop/충남,대전, 세종 무인교통단속카메라/seosan.csv",encoding='cp949')
TAEAN_CCTV = pd.read_csv("C:/Users/user/Desktop/충남,대전, 세종 무인교통단속카메라/taean.csv",encoding='cp949')
YESAN_CCTV = pd.read_csv("C:/Users/user/Desktop/충남,대전, 세종 무인교통단속카메라/yesan.csv",encoding='cp949')

|Column|Description|
|:---|:---|
|RECV_DEPT_NM|접수 부서|
|RECV_CPLT_DM|접수 완료 일시|
|NPA_CL|경찰청 구분(코드 번호)|
|EVT_STAT_CD|사건 상태 코드|
|EVT_CL_CD|사건 종별 코드|
|RPTER_SEX|성별(남:1, 여:2, 불상:3)|
|HPPN_PNU_ADDR|사건 발생 주소|
|HPPN_X|주소 좌표(경도)|
|HPPN_Y|주소 좌표(위도)|
|SME_EVT_Y|동일 사건 여부|

### NPA Column

|Column|Description|
|:---|:---|
|RECV_CPLT_DT|접수 완료 일자|
|RECV_CPLT_TM|접수 완료 시간|

# 2. Data 전처리

## 1. KP_2020

In [423]:
### 필요없는 항목 제거 => 경찰청 이름, 시간, 사건 발생 주소
KP_2020.drop(['RECV_DEPT_NM','RECV_CPLT_DM','HPPN_PNU_ADDR'], axis=1, inplace=True)
KP_2020

NPA_CL  EVT_STAT_CD  EVT_CL_CD  RPTER_SEX      HPPN_X     HPPN_Y  \
0          19           10        305        1.0  126.598345  36.341537   
1          13           10        601        3.0  127.404663  36.341685   
2          13           10        601        1.0  127.404663  36.341685   
3          19           10        606        3.0  126.524980  36.474390   
4          19           10        401        2.0  127.137160  36.826718   
...       ...          ...        ...        ...         ...        ...   
77072      19           10        302        2.0  126.571002  36.387791   
77073      19           10        501        3.0         NaN        NaN   
77074      19           10        604        3.0  127.000391  36.788411   
77075      13           10        302        2.0  127.373351  36.348051   
77076      13           10        601        3.0         NaN        NaN   

      SME_EVT_YN  
0              Y  
1            NaN  
2            NaN  
3              N  
4            NaN  
...          ...  
77072        NaN  
77073        NaN  
77074        NaN  
77075          Y  
77076        NaN  

[77077 rows x 7 columns]

In [424]:
KP_2020.isnull().sum()

NPA_CL             0
EVT_STAT_CD        0
EVT_CL_CD          0
RPTER_SEX       2419
HPPN_X         16764
HPPN_Y         16764
SME_EVT_YN     63635
dtype: int64

In [425]:
### NULL 값 많은것도 그냥 삭제
KP_2020.drop(['RPTER_SEX','SME_EVT_YN'], axis=1, inplace=True)
KP_2020.head()

NPA_CL  EVT_STAT_CD  EVT_CL_CD      HPPN_X     HPPN_Y
0      19           10        305  126.598345  36.341537
1      13           10        601  127.404663  36.341685
2      13           10        601  127.404663  36.341685
3      19           10        606  126.524980  36.474390
4      19           10        401  127.137160  36.826718

## 2. KP_2021

In [426]:
### 필요없는 항목 제거 => 경찰청 이름, 시간, 사건 발생 주소, 주소 좌표

KP_2021.drop(['RECV_DEPT_NM','RECV_CPLT_DM','HPPN_PNU_ADDR'], axis=1, inplace=True)

In [427]:
KP_2021.isnull().sum()

NPA_CL               0
EVT_STAT_CD          0
EVT_CL_CD            0
RPTER_SEX        23894
HPPN_X          694401
HPPN_Y          694401
SME_EVT_YN     2079123
dtype: int64

In [428]:
### NULL 값 많은것도 그냥 삭제
KP_2021.drop(['RPTER_SEX','SME_EVT_YN'], axis=1, inplace=True)
KP_2021.head()

NPA_CL  EVT_STAT_CD  EVT_CL_CD      HPPN_X     HPPN_Y
0      13           10        604  127.373676  36.350975
1      13           10        201  127.339018  36.347420
2      13           10        601  127.404663  36.341685
3      13           10        601  127.404663  36.341685
4      13           10        308  127.420455  36.330413

## 3. NPA_2020

In [429]:
### 필요없는 항목 제거 => 사건 날짜, 접수 완료 시간, 발생 주소, 발생 좌표, 동일 사건 여부
NPA_2020.drop(['RECV_CPLT_DT','HPPN_OLD_ADDR','SME_EVT_YN','RPTER_SEX','RECV_CPLT_TM'], axis=1, inplace=True)
NPA_2020

NPA_CL  EVT_STAT_CD  EVT_CL_CD      HPPN_X     HPPN_Y
0            13           10        501  127.409270  36.333010
1            13           10        501  127.421295  36.325575
2            13           10        501  127.404663  36.341685
3            13           10        601    0.000000   0.000000
4            13           10        601  127.404663  36.341685
...         ...          ...        ...         ...        ...
1178239      19           10        501         NaN        NaN
1178240      13           10        601  127.404663  36.341685
1178241      19           10        601         NaN        NaN
1178242      19            5        301  126.516040  36.305619
1178243      13           10        601  127.404663  36.341685

[1178244 rows x 5 columns]

## CCTV

#### 1. 아산

In [430]:
ASAN_CCTV.head()

무인교통단속카메라관리번호   시도명 시군구명  도로종류 도로노선번호 도로노선명  도로노선방향            소재지도로명주소  \
0      NEXPA_01  충청남도  아산시  일반국도    48번   실옥로     NaN     충청남도 아산시 실옥로 48   
1      NEXPA_02  충청남도  아산시  일반국도   374번   시민로     NaN    충청남도 아산시 시민로 374   
2      NEXPA_03  충청남도  아산시  일반국도  1627번  온천대로     NaN  충청남도 아산시 온천대로 1627   
3      NEXPA_04  충청남도  아산시  일반국도   133번   번영로     NaN    충청남도 아산시 번영로 133   
4      NEXPA_05  충청남도  아산시  일반국도   101번   번영로     NaN    충청남도 아산시 번영로 101   

   소재지지번주소         위도  ...        설치장소 단속구분  제한속도  단속구간위치구분  과속단속구간길이  보호구역구분  \
0      NaN  36.787433  ...    천도초등학교주변    4     0       NaN       NaN     2.0   
1      NaN  36.781864  ...  온양관광호텔 사거리    4     0       NaN       NaN     NaN   
2      NaN  36.778944  ...    동신초등학교주변    4     0       NaN       NaN     2.0   
3      NaN  36.786259  ...     아고오거리주변    4     0       NaN       NaN     NaN   
4      NaN  36.786764  ...    시민로사거리주변    4     0       NaN       NaN     NaN   

         설치년도       관리기관명      관리기관전화번호     데이터기준일자  
0  2006-09-01  아산시청 교통행정과  041-540-2953  2020-09-17  
1  2006-09-01  아산시청 교통행정과  041-540-2953  2020-09-17  
2  2006-09-01  아산시청 교통행정과  041-540-2953  2020-09-17  
3  2006-09-01  아산시청 교통행정과  041-540-2953  2020-09-17  
4  2006-09-01  아산시청 교통행정과  041-540-2953  2020-09-17  

[5 rows x 21 columns]

In [431]:
ASAN_CCTV.columns

Index(['무인교통단속카메라관리번호', '시도명', '시군구명', '도로종류', '도로노선번호', '도로노선명', '도로노선방향',
       '소재지도로명주소', '소재지지번주소', '위도', '경도', '설치장소', '단속구분', '제한속도', '단속구간위치구분',
       '과속단속구간길이', '보호구역구분', '설치년도', '관리기관명', '관리기관전화번호', '데이터기준일자'],
      dtype='object')

In [432]:
ASAN_CCTV.drop(['무인교통단속카메라관리번호','도로노선번호','도로노선방향','시도명','시군구명','도로종류','소재지도로명주소','소재지지번주소','단속구분','제한속도','단속구간위치구분','과속단속구간길이','보호구역구분','설치년도','관리기관명','관리기관전화번호','데이터기준일자'], axis=1, inplace=True)
ASAN_CCTV.head()

도로노선명         위도          경도        설치장소
0   실옥로  36.787433  126.992743    천도초등학교주변
1   시민로  36.781864  127.000437  온양관광호텔 사거리
2  온천대로  36.778944  127.018453    동신초등학교주변
3   번영로  36.786259  127.005606     아고오거리주변
4   번영로  36.786764  127.002256    시민로사거리주변

In [433]:
ASAN_CCTV.value_counts

<bound method DataFrame.value_counts of     도로노선명         위도          경도            설치장소
0     실옥로  36.787433  126.992743        천도초등학교주변
1     시민로  36.781864  127.000437      온양관광호텔 사거리
2    온천대로  36.778944  127.018453        동신초등학교주변
3     번영로  36.786259  127.005606         아고오거리주변
4     번영로  36.786764  127.002256        시민로사거리주변
..    ...        ...         ...             ...
116   용화로  36.776829  127.008517  용화동 아프리카카페 사거리
117   모산로  36.772828  127.046870   배방 태성연립 앞 사거리
118  신정호길  36.761510  126.966909     신창 북한강쭈꾸미 앞
119   시민로  36.785013  127.001220   온양원도심 양성평등거리1
120   시민로  36.785433  127.000391   온양원도심 양성평등거리2

[121 rows x 4 columns]>

In [434]:
ASAN_CCTV.dropna(axis='index',how='any',inplace=True)

In [435]:
ASAN_CCTV.isnull().sum()

도로노선명    0
위도       0
경도       0
설치장소     0
dtype: int64

#### 2. 보령

In [436]:
BORYEONG_CCTV.columns

Index(['무인교통단속카메라관리번호', '시도명', '시군구명', '도로종류', '도로노선번호', '도로노선명', '도로노선방향',
       '소재지도로명주소', '소재지지번주소', '위도', '경도', '설치장소', '단속구분', '제한속도', '단속구간위치구분',
       '과속단속구간길이', '보호구역구분', '설치연도', '관리기관명', '관리기관전화번호', '데이터기준일자'],
      dtype='object')

In [437]:
BORYEONG_CCTV.drop(['도로노선번호','도로노선방향','시도명','시군구명','도로종류','소재지도로명주소','소재지지번주소','단속구분','제한속도','단속구간위치구분','과속단속구간길이','보호구역구분','설치연도','관리기관명','관리기관전화번호','데이터기준일자','무인교통단속카메라관리번호'], axis=1, inplace=True)
BORYEONG_CCTV.head()

도로노선명         위도          경도        설치장소
0   번영로  36.351535  126.593371         사거리
1   중앙로  36.351064  126.594728         사거리
2   중앙로  36.350414  126.596749         사거리
3   대흥로  36.348748  126.596251         사거리
4   중앙로  36.348686  126.598536  일방통행종점 삼거리

In [438]:
BORYEONG_CCTV.isnull().sum()

도로노선명    0
위도       0
경도       0
설치장소     0
dtype: int64

In [439]:
BORYEONG_CCTV.dropna(axis='index',how='any',inplace=True)
BORYEONG_CCTV.isnull().sum()

도로노선명    0
위도       0
경도       0
설치장소     0
dtype: int64

#### 3. 천안

In [440]:
CHEONAN_CCTV.drop(['도로노선방향','도로노선번호','시도명','시군구명','도로종류','소재지도로명주소','소재지지번주소','단속구분','제한속도','단속구간위치구분','과속단속구간길이','보호구역구분','설치연도','관리기관명','관리기관전화번호','데이터기준일자','무인교통단속카메라관리번호'], axis=1, inplace=True)
CHEONAN_CCTV.head()

도로노선명         위도          경도      설치장소
0   봉정로  36.810381  127.140578   서부역 삼거리
1   봉정로  36.813499  127.141843  성정사거리 부근
2   봉정로  36.818320  127.143078  봉정사거리 부근
3   월봉로  36.798973  127.118461   맘마마트 부근
4   월봉로  36.802538  127.118051    용암마을주변

In [441]:
CHEONAN_CCTV.isnull().sum()

도로노선명    0
위도       0
경도       0
설치장소     0
dtype: int64

In [442]:
CHEONAN_CCTV.dropna(axis='index',how='any',inplace=True)
CHEONAN_CCTV.isnull().sum()

도로노선명    0
위도       0
경도       0
설치장소     0
dtype: int64

#### 4.청양

In [443]:
CHUNGYANG_CCTV.drop(['도로노선방향','도로노선번호','시도명','시군구명','도로종류','소재지도로명주소','소재지지번주소','단속구분','제한속도','단속구간위치구분','과속단속구간길이','보호구역구분','설치연도','관리기관명','관리기관전화번호','데이터기준일자','무인교통단속카메라관리번호'], axis=1, inplace=True)
CHUNGYANG_CCTV.head()

도로노선명         위도          경도       설치장소
0   중앙로  36.451900  126.802982  시외버스터미널 앞
1  칠갑산로  36.451221  126.801630   문화체육센터 앞
2   중앙로  36.451017  126.803471     삼성의원 앞
3   중앙로  36.450600  126.802781    새마을금고 앞
4  칠갑산로  36.454261  126.804666     청양초 후문

In [444]:
CHUNGYANG_CCTV.isnull().sum()

도로노선명    0
위도       0
경도       0
설치장소     0
dtype: int64

#### 5. 대전

In [445]:
DAEJEON_CCTV.head()

무인교통단속카메라관리번호    시도명 시군구명 도로종류  도로노선번호 도로노선명  도로노선방향  소재지도로명주소  \
0         F5705  대전광역시  유성구   시도     NaN  한밭대로       2       NaN   
1         F5706  대전광역시   서구   시도     NaN  대덕대로       2       NaN   
2         F5707  대전광역시   서구   시도     NaN  동서대로       2       NaN   
3         F6307  대전광역시   서구   시도     NaN   계백로       1       NaN   
4         F6308  대전광역시   서구   시도     NaN  도안동로       2       NaN   

              소재지지번주소         위도  ...             설치장소 단속구분  제한속도  단속구간위치구분  \
0   대전광역시 유성구 장대동 339  36.364138  ...            장대네거리    2    50       NaN   
1   대전광역시 서구 둔산동 1535  36.348777  ...           큰마을네거리    2    50       NaN   
2   대전광역시 서구 도안동 1512  36.332055  ...  하나은행 도안지점 앞 네거리    2    50       NaN   
3  대전광역시 서구 정림동 293-4  36.307683  ...            정림삼거리    2    60       NaN   
4   대전광역시 서구 도안동 1472  36.312888  ...            용소삼거리    2    50       NaN   

   과속단속구간길이  보호구역구분    설치연도  관리기관명 관리기관전화번호     데이터기준일자  
0       NaN     NaN  2010.0  대전광역시  042-120  2022-12-15  
1       NaN     NaN  2010.0  대전광역시  042-120  2022-12-15  
2       NaN     NaN  2010.0  대전광역시  042-120  2022-12-15  
3       NaN     NaN  2011.0  대전광역시  042-120  2022-12-15  
4       NaN     NaN  2011.0  대전광역시  042-120  2022-12-15  

[5 rows x 21 columns]

In [446]:
DAEJEON_CCTV.columns

Index(['무인교통단속카메라관리번호', '시도명', '시군구명', '도로종류', '도로노선번호', '도로노선명', '도로노선방향',
       '소재지도로명주소', '소재지지번주소', '위도', '경도', '설치장소', '단속구분', '제한속도', '단속구간위치구분',
       '과속단속구간길이', '보호구역구분', '설치연도', '관리기관명', '관리기관전화번호', '데이터기준일자'],
      dtype='object')

In [447]:
DAEJEON_CCTV.drop(['도로노선방향','도로노선번호','시도명','시군구명','도로종류','소재지도로명주소','소재지지번주소','단속구분','제한속도','단속구간위치구분','과속단속구간길이','보호구역구분','설치연도','관리기관명','관리기관전화번호','데이터기준일자','무인교통단속카메라관리번호'], axis=1, inplace=True)
DAEJEON_CCTV.head()

도로노선명         위도          경도             설치장소
0  한밭대로  36.364138  127.338138            장대네거리
1  대덕대로  36.348777  127.376944           큰마을네거리
2  동서대로  36.332055  127.341083  하나은행 도안지점 앞 네거리
3   계백로  36.307683  127.364342            정림삼거리
4  도안동로  36.312888  127.350083            용소삼거리

In [448]:
DAEJEON_CCTV.dropna(axis='index',how='any',inplace=True)

In [449]:
DAEJEON_CCTV.isnull().sum()

도로노선명    0
위도       0
경도       0
설치장소     0
dtype: int64

#### 6. 당진

In [450]:
DANGJIN_CCTV.drop(['도로노선방향','도로노선번호','시도명','시군구명','도로종류','소재지도로명주소','소재지지번주소','단속구분','제한속도','단속구간위치구분','과속단속구간길이','보호구역구분','설치연도','관리기관명','관리기관전화번호','데이터기준일자','무인교통단속카메라관리번호'], axis=1, inplace=True)
DANGJIN_CCTV.head()

도로노선명         위도          경도                    설치장소
0  남산공원길  36.891894  126.631523   당진초등학교(당진시 읍내동 167-8)
1    밤절로        NaN         NaN  계성초등학교(당진시 읍내동 735-11)
2   대호만로        NaN         NaN   탑동초등학교(당진시 채운동 456-1)
3   대덕1로        NaN         NaN    대덕초등학교(당진시 대덕동 1919)
4   대덕1로        NaN         NaN    대덕초등학교(당진시 대덕동 1919)

In [451]:
DANGJIN_CCTV.isnull().sum()

도로노선명     0
위도       10
경도       10
설치장소      0
dtype: int64

In [452]:
DANGJIN_CCTV.dropna(axis='index',how='any',inplace=True)
DANGJIN_CCTV.isnull().sum()

도로노선명    0
위도       0
경도       0
설치장소     0
dtype: int64

#### 7. 금산

In [453]:
GEOMSAN_CCTV.drop(['도로노선방향','도로노선번호','시도명','시군구명','도로종류','소재지도로명주소','소재지지번주소','단속구분','제한속도','단속구간위치구분','과속단속구간길이','보호구역구분','설치연도','관리기관명','관리기관전화번호','데이터기준일자','무인교통단속카메라관리번호'], axis=1, inplace=True)
GEOMSAN_CCTV.head()

도로노선명         위도          경도          설치장소
0   비범로  36.104751  127.491272        중부시장 옆
1   인삼로  36.102834  127.490645     군농협중앙회 4가
2  비호산로  36.109932  127.489056  주공1차아파트 입구 앞
3   인삼로  36.103116  127.488573      금산우체국 4가
4   비호로  36.107215  127.489674   금산행정복지센터 4가

In [454]:
GEOMSAN_CCTV.isnull().sum()

도로노선명    0
위도       0
경도       0
설치장소     0
dtype: int64

#### 8. 세종

In [455]:
SEJONG_CCTV.head()

무인교통단속카메라관리번호      시도명 시군구명  도로종류  도로노선번호  도로노선명  도로노선방향 소재지도로명주소  \
0         F7219  세종특별자치시   없음  일반국도     1.0    세종로       2      NaN   
1         F7460  세종특별자치시   없음   지방도   327.0  의당전의로       1      NaN   
2         F7461  세종특별자치시   없음   지방도   327.0  의당전의로       1      NaN   
3         F7641  세종특별자치시   없음  일반국도     1.0    세종로       1      NaN   
4         F7642  세종특별자치시   없음  일반국도     1.0    세종로       1      NaN   

                 소재지지번주소         위도          경도  \
0  세종특별자치시 소정면 대곡리 322-2  36.728495  127.165088   
1  세종특별자치시 장군면 송학리 113-4  36.527810  127.184503   
2  세종특별자치시 장군면 송학리 113-4  36.527810  127.184503   
3  세종특별자치시 조치원읍 침산리 34-5  36.603117  127.289840   
4   세종특별자치시 소정면 운당리 32-5  36.697907  127.152629   

                                     설치장소  단속구분  제한속도  설치연도       관리기관명  \
0       세종특별자치시 소정면 세종로 4679, (은산제2주유소 앞)     1    80  2013  세종특별자치시경찰청   
1        세종특별자치시 장군면 의당전의로 송학1리 마을입구 공주방향     1    60  2014  세종특별자치시경찰청   
2        세종특별자치시 장군면 의당전의로 송학1리 마을입구 세종방향     1    60  2014  세종특별자치시경찰청   
3  세종특별자치시 조치원읍 세종로 욱일사거리(신봉초교 앞) (대전→천안)     2    50  2014  세종특별자치시경찰청   
4            세종특별자치시 소정면 세종로 무궁화@앞(천안 방향)     2    80  2014  세종특별자치시경찰청   

       관리기관전화번호     데이터기준일자  
0  044-559-2151  2022-05-20  
1  044-559-2151  2022-05-20  
2  044-559-2151  2022-05-20  
3  044-559-2151  2022-05-20  
4  044-559-2151  2022-05-20

In [456]:
SEJONG_CCTV.columns

Index(['무인교통단속카메라관리번호', '시도명', '시군구명', '도로종류', '도로노선번호', '도로노선명', '도로노선방향',
       '소재지도로명주소', '소재지지번주소', '위도', '경도', '설치장소', '단속구분', '제한속도', '설치연도',
       '관리기관명', '관리기관전화번호', '데이터기준일자'],
      dtype='object')

In [457]:
SEJONG_CCTV.drop(['무인교통단속카메라관리번호', '시도명', '시군구명', '도로종류', '도로노선번호', '도로노선방향',
       '소재지도로명주소', '소재지지번주소', '단속구분', '제한속도', '설치연도',
       '관리기관명', '관리기관전화번호', '데이터기준일자'], axis=1, inplace=True)
SEJONG_CCTV.head()

도로노선명         위도          경도                                    설치장소
0    세종로  36.728495  127.165088       세종특별자치시 소정면 세종로 4679, (은산제2주유소 앞)
1  의당전의로  36.527810  127.184503        세종특별자치시 장군면 의당전의로 송학1리 마을입구 공주방향
2  의당전의로  36.527810  127.184503        세종특별자치시 장군면 의당전의로 송학1리 마을입구 세종방향
3    세종로  36.603117  127.289840  세종특별자치시 조치원읍 세종로 욱일사거리(신봉초교 앞) (대전→천안)
4    세종로  36.697907  127.152629            세종특별자치시 소정면 세종로 무궁화@앞(천안 방향)

In [458]:
SEJONG_CCTV.isnull().sum()

도로노선명    0
위도       0
경도       0
설치장소     0
dtype: int64

#### 9. 서천

In [459]:
SEOCHEON_CCTV.drop(['도로노선방향','도로노선번호','시도명','시군구명','도로종류','소재지도로명주소','소재지지번주소','단속구분','제한속도','단속구간위치구분','과속단속구간길이','보호구역구분','설치연도','관리기관명','관리기관전화번호','데이터기준일자','무인교통단속카메라관리번호'], axis=1, inplace=True)
SEOCHEON_CCTV.head()

도로노선명         위도          경도       설치장소
0    중로  36.078422  126.691358    서천군청 입구
1    중로  36.077096  126.690699      군청사거리
2    중로  36.076513  126.693599    서천중학교입구
3    중로  36.077547  126.689105  서천초등학교 입구
4    중로  36.079068  126.686136      다사랑입구

In [460]:
SEOCHEON_CCTV.isnull().sum()

도로노선명    0
위도       0
경도       0
설치장소     0
dtype: int64

#### 10. 서산

In [461]:
SEOSAN_CCTV.drop(['도로노선방향','도로노선번호','시도명','시군구명','도로종류','소재지도로명주소','소재지지번주소','단속구분','제한속도','단속구간위치구분','과속단속구간길이','보호구역구분','설치연도','관리기관명','관리기관전화번호','데이터기준일자','무인교통단속카메라관리번호'], axis=1, inplace=True)
SEOSAN_CCTV.head()

도로노선명         위도          경도                         설치장소
0   서해로  36.770225  126.434310                    서해로 법원삼거리
1  잠홍2길  36.788711  126.485470        잠홍2길 21 잠홍 장례식장 앞 사거리
2   서해로  36.785559  126.475696                    서해로 애향삼거리
3   충의로  36.877264  126.439599  지곡면 충의로 화천교차로 (화천슈퍼 앞) 대산방향
4   양열로  36.761317  126.453227           석남동 양열로 석지사거리 태안방향

In [462]:
SEOSAN_CCTV.isnull().sum()

도로노선명    0
위도       0
경도       0
설치장소     0
dtype: int64

#### 11. 태안

In [463]:
TAEAN_CCTV.drop(['도로노선방향','도로노선번호','시도명','시군구명','도로종류','소재지도로명주소','소재지지번주소','단속구분','제한속도','단속구간위치구분','과속단속구간길이','보호구역구분','설치연도','관리기관명','관리기관전화번호','데이터기준일자','무인교통단속카메라관리번호'], axis=1, inplace=True)
TAEAN_CCTV.head()

도로노선명         위도          경도 설치장소
0   중앙로  36.754050  126.296765  중앙로
1   중앙로  36.753216  126.298963  중앙로
2   중앙로  36.748852  126.296810  중앙로
3   중앙로  36.754363  126.294761  중앙로
4   중앙로  36.754057  126.297036  중앙로

In [464]:
TAEAN_CCTV.isnull().sum()

도로노선명    0
위도       0
경도       0
설치장소     0
dtype: int64

#### 12. 예산

In [465]:
YESAN_CCTV.drop(['도로노선방향','도로노선번호','시도명','시군구명','도로종류','소재지도로명주소','소재지지번주소','단속구분','제한속도','단속구간위치구분','과속단속구간길이','보호구역구분','설치연도','관리기관명','관리기관전화번호','데이터기준일자','무인교통단속카메라관리번호'], axis=1, inplace=True)
YESAN_CCTV.head()

도로노선명         위도          경도     설치장소
0   봉운로  36.705838  126.669772     덕산초교
1   역전로  36.682523  126.829277    주교사거리
2  금오대로  36.693884  126.831997  종합버스터미널
3   임성로  36.679840  126.849771    삼선당약국
4   임성로  36.678431  126.849570     광덕주단

In [466]:
YESAN_CCTV.isnull().sum()

도로노선명    0
위도       0
경도       0
설치장소     0
dtype: int64

#### column 통일

In [467]:
KP_2020.columns

Index(['NPA_CL', 'EVT_STAT_CD', 'EVT_CL_CD', 'HPPN_X', 'HPPN_Y'], dtype='object')

In [468]:
KP_2021.columns

Index(['NPA_CL', 'EVT_STAT_CD', 'EVT_CL_CD', 'HPPN_X', 'HPPN_Y'], dtype='object')

In [469]:
NPA_2020.columns

Index(['NPA_CL', 'EVT_STAT_CD', 'EVT_CL_CD', 'HPPN_X', 'HPPN_Y'], dtype='object')

### 3. 데이터 병합

In [470]:
# 원본데이터가 바뀌는것을 미연에 방지하기위해 다른 데이터에 복사 후 진행
k_2020 = KP_2020.copy()
k_2021 = KP_2021.copy()
n_2020 = NPA_2020.copy()

# 교통사고건만 추출 후 변수에 저장 
k_2020 = k_2020[k_2020['EVT_CL_CD']==401]
k_2021 = k_2021[k_2021['EVT_CL_CD']==401]
n_2020 = n_2020[n_2020['EVT_CL_CD']==401]

In [471]:
print('k_2020 csv 파일 교통사고 건 수 = ', len(k_2020))
print('k_2021 csv 파일 교통사고 건 수 = ', len(k_2021))
print('n_2020 csv 파일 교통사고 건 수 = ', len(n_2020))
print('\n전체 교통사고 건 수  = ', len(k_2020)+len(k_2021)+len(n_2020))

k_2020 csv 파일 교통사고 건 수 =  6207
k_2021 csv 파일 교통사고 건 수 =  164582
n_2020 csv 파일 교통사고 건 수 =  93659

전체 교통사고 건 수  =  264448


### CCTV 병합

#### 중복값 제거

In [472]:
#전체
CCTV = pd.concat([ASAN_CCTV,BORYEONG_CCTV,CHEONAN_CCTV,CHUNGYANG_CCTV,DAEJEON_CCTV,DANGJIN_CCTV,GEOMSAN_CCTV,SEJONG_CCTV, 
                 SEOCHEON_CCTV,SEOSAN_CCTV,TAEAN_CCTV,YESAN_CCTV  ])
#충남
CH_CCTV = pd.concat([ASAN_CCTV,BORYEONG_CCTV,CHEONAN_CCTV,CHUNGYANG_CCTV,DANGJIN_CCTV,GEOMSAN_CCTV,
                 SEOCHEON_CCTV,SEOSAN_CCTV,TAEAN_CCTV,YESAN_CCTV  ])
#대전
DJ_CCTV = pd.concat([DAEJEON_CCTV])
#세종
SJ_CCTV = pd.concat([SEJONG_CCTV])

In [473]:
All = pd.concat([k_2020,k_2021,n_2020])
All = All.drop_duplicates()
print('중복값 제거 후 전체 교통사고 건수 : ', len(All))

중복값 제거 후 전체 교통사고 건수 :  191655


In [474]:
All = All.fillna({'HPPN_X':All['HPPN_X'].mean()})
All = All.fillna({'HPPN_Y':All['HPPN_Y'].mean()})
All.isnull().sum()

NPA_CL         0
EVT_STAT_CD    0
EVT_CL_CD      0
HPPN_X         0
HPPN_Y         0
dtype: int64

|Column|서단(경도)|동단(경도)|북단(위도)|남단(위도)|
|:---|:------|:------|:------|:------|
|대전|127.14|127.33|36.30|36.10|
|세종|127.1|127.23|36.43|36.24|
|충남|125.3221|127.3831|37.0344|35.5830|

## 충남 대전 세종지역 추출 

In [475]:
#대전
DJPC = All[All['NPA_CL']== 13]

#충남
CHPC = All[All['NPA_CL']== 19]

#세종
SJPC = All[All['NPA_CL']== 31]

print('모든 청 관리 교통사고 건수 : ', len(All))
print('대전청 관리 교통사고 건수 : ', len(DJPC))
print('충남청 관리 교통사고 건수 : ', len(CHPC))
print('세종청 관리 교통사고 건수 : ', len(SJPC))

모든 청 관리 교통사고 건수 :  191655
대전청 관리 교통사고 건수 :  68413
충남청 관리 교통사고 건수 :  114161
세종청 관리 교통사고 건수 :  7463


#### 지역 추출 변환 함수

In [476]:
# 원하고자 하는 지역 외의 데이터 삭제 및 뽑기
# 변환데이터, 원본데이터, 동단경도, 서단경도, 남단위도, 북단위도

# 대전, 세종에 해당 
def translocation(TransData, Data,  Eastlongtitude, Westlongtitude,  Southlatitude, Northlatitude):

     #동단 삭제 
    TransData = Data.drop(Data[(Data['HPPN_X']>Eastlongtitude)].index)
    
    #서단 삭제
    TransData = TransData.drop(TransData[(TransData['HPPN_X']<Westlongtitude)].index)
    
    #남단 삭제 
    TransData = TransData.drop(TransData[(TransData['HPPN_Y']<Southlatitude)].index)
    
    #북단 삭제 
    TransData =TransData.drop(TransData[(TransData['HPPN_Y']>Northlatitude)].index)
    
    return TransData

# 원하고자 하는 지역 만 뽑기(충남의 경우만 해당)

def translocation_CH(TransData, Data, Eastlongtitude, Westlongtitude, Southlatitude, Northlatitude):
    
    TransData = Data.drop(Data[(Data.HPPN_X < Eastlongtitude) & (Data.HPPN_X > Westlongtitude) &
                              (Data.HPPN_Y > Southlatitude) & (Data.HPPN_Y < Northlatitude)
                              ].index)
    return TransData

def duplicated_location(Data):
    print('중복 처리 전 교통사고 건수: ', len(Data))
    Data[Data.duplicated(['HPPN_X','HPPN_Y'])==True]
    Data = Data.drop_duplicates(['HPPN_X','HPPN_Y'])
    print('중복 처리 후 교통사고 건수 : ', len(Data))
    
    return Data

### 충남

In [477]:
# 충남 - 대전 
Chungnam = pd.DataFrame()

Chungnam = translocation_CH(Chungnam, All, 127.55529876331815, 127.25007536223731, 36.186117749974656, 36.49779017085958 )
print('충남지역에서 대전지역에 해당하는 데이터 제거 후 개수 : ', len(Chungnam))

# 충남 - 세종   127.23, 127.1, 36.24, 36.43)
Chungnam = translocation_CH(Chungnam, Chungnam, 127.40746822278025, 127.12869800443497, 36.41194803285692, 36.70492)
print('충남지역에서 세종지역에 해당하는 데이터 제거 후 개수 : ', len(Chungnam))

Chungnam = translocation(Chungnam,Chungnam, 127.62692947603428,126.13148910811715,35.98803519542761,37.056951491204124 )

충남지역에서 대전지역에 해당하는 데이터 제거 후 개수 :  115403
충남지역에서 세종지역에 해당하는 데이터 제거 후 개수 :  105032


In [478]:
Chungnam = duplicated_location(Chungnam)

중복 처리 전 교통사고 건수:  103893
중복 처리 후 교통사고 건수 :  103129


### 대전

In [479]:
DAEJEON = pd.DataFrame()
DAEJEON = translocation(DAEJEON, All, 127.55529876331815, 127.25007536223731, 36.186117749974656, 36.49779017085958)
print('대전지역 교통사고 건수 : ', len(DAEJEON))

대전지역 교통사고 건수 :  72981


In [480]:
DAEJEON = duplicated_location(DAEJEON)

중복 처리 전 교통사고 건수:  72981
중복 처리 후 교통사고 건수 :  70712


### 세종

In [481]:
SEJONG = pd.DataFrame()
SEJONG = translocation(SEJONG, All, 127.40746822278025, 127.12869800443497, 36.41194803285692, 36.70492)
print('세종지역 교통사고 건수 : ', len(SEJONG))

세종지역 교통사고 건수 :  15588


In [482]:
SEJONG = duplicated_location(SEJONG)

중복 처리 전 교통사고 건수:  15588
중복 처리 후 교통사고 건수 :  14854


### 충남, 대전, 세종 병합 = 전체교통사고확인

In [483]:
# 병합
CDS= pd.concat([SEJONG,DAEJEON,Chungnam])
print('중복 처리 전 전체 교통사고 건수: ', len(CDS))

# 중복 값, 개수 확인
CDS[CDS.duplicated(['HPPN_X','HPPN_Y'])==True]

중복 처리 전 전체 교통사고 건수:  188695


NPA_CL  EVT_STAT_CD  EVT_CL_CD      HPPN_X     HPPN_Y
426          13           10        401  127.386404  36.435266
1387         31           10        401  127.293270  36.482200
1943         31           10        401  127.262514  36.480017
2124         31           10        401  127.262247  36.479612
3541         31           10        401  127.268839  36.468472
...         ...          ...        ...         ...        ...
1172176      19           10        401  127.258070  36.489040
1172705      13           10        401  127.404595  36.445130
1172823      19           10        401  127.290190  36.479180
1173190      19           10        401  127.296364  36.475845
1177827      13           10        401  127.394790  36.438470

[5291 rows x 5 columns]

In [484]:
# 중복 값 제거, (HPPN_X와 HPPN_Y둘다 같아야 제거)
CDS = duplicated_location(CDS)

중복 처리 전 교통사고 건수:  188695
중복 처리 후 교통사고 건수 :  183404


In [485]:
CDS.isnull().sum()

NPA_CL         0
EVT_STAT_CD    0
EVT_CL_CD      0
HPPN_X         0
HPPN_Y         0
dtype: int64

### 충남, 대전, 세종 지역에서 각 청이 관리하는 사건만 추출

In [486]:
DJPC = DAEJEON[DAEJEON['NPA_CL']== 13]
SJPC = SEJONG[SEJONG['NPA_CL']== 31]
CHPC = Chungnam[Chungnam['NPA_CL']== 19]

print('대전청 교통사고 관리 건 수 ', len(DJPC))
print('세종청 교통사고 관리 건 수 ', len(SJPC))
print('충남청 교통사고 관리 건 수 ', len(CHPC))

대전청 교통사고 관리 건 수  65623
세종청 교통사고 관리 건 수  7155
충남청 교통사고 관리 건 수  102638


### 2020년 데이터

In [487]:
TA_2020 = pd.concat([k_2020, n_2020])
len(TA_2020)

99866

In [488]:
TA_2020 = duplicated_location(TA_2020)

중복 처리 전 교통사고 건수:  99866
중복 처리 후 교통사고 건수 :  72985


In [489]:
TA_2020 = TA_2020.fillna({'HPPN_X':TA_2020['HPPN_X'].mean()})
TA_2020 = TA_2020.fillna({'HPPN_Y':TA_2020['HPPN_Y'].mean()})
TA_2020.isnull().sum()

NPA_CL         0
EVT_STAT_CD    0
EVT_CL_CD      0
HPPN_X         0
HPPN_Y         0
dtype: int64

#### 2020년 충남

In [490]:
# 충남 - 대전 
CH_2020 = pd.DataFrame()
CH_2020 = translocation_CH(CH_2020, TA_2020, 127.55529876331815, 127.25007536223731, 36.186117749974656, 36.49779017085958 )
print('2020년 충남지역에서 대전지역에 해당하는 데이터 제거 후 개수 : ', len(CH_2020))

# 충남 - 세종   127.23, 127.1, 36.24, 36.43)
CH_2020 = translocation_CH(CH_2020, CH_2020, 127.40746822278025, 127.12869800443497, 36.41194803285692, 36.70492)
print('2020년 충남지역에서 세종지역에 해당하는 데이터 제거 후 개수 : ', len(CH_2020))

CH_2020 = translocation(CH_2020,CH_2020, 127.62692947603428,126.13148910811715,35.98803519542761,37.056951491204124 )
print('2020년 충남지역에서 교통사고 건수 : ', len(CH_2020))

2020년 충남지역에서 대전지역에 해당하는 데이터 제거 후 개수 :  44878
2020년 충남지역에서 세종지역에 해당하는 데이터 제거 후 개수 :  41189
2020년 충남지역에서 교통사고 건수 :  40888


#### 2020년 대전

In [491]:
DJ_2020 = pd.DataFrame()
DJ_2020 = translocation(DJ_2020, TA_2020, 127.55529876331815, 127.25007536223731, 36.186117749974656, 36.49779017085958)
print('2020년 충남지역에서 교통사고 건수 : ', len(DJ_2020))

2020년 충남지역에서 교통사고 건수 :  27743


#### 2020년 세종

In [492]:
SJ_2020 = pd.DataFrame()
SJ_2020 = translocation(SJ_2020, TA_2020, 127.40746822278025, 127.12869800443497, 36.41194803285692, 36.70492)
print('2020년 충남지역에서 교통사고 건수 : ', len(SJ_2020))

2020년 충남지역에서 교통사고 건수 :  5650


### 2020년 충남, 대전, 세종 교통사고 건수 합

In [493]:
TA_2020_total = pd.concat([CH_2020, DJ_2020, SJ_2020])
len(TA_2020)

72985

In [494]:
TA_2020_total = duplicated_location(TA_2020_total)

중복 처리 전 교통사고 건수:  74281
중복 처리 후 교통사고 건수 :  72280


### 2020년 각 경찰청 별로 관리한 교통사고 건수

In [495]:
DJPC_2020 = DJ_2020[DJ_2020['NPA_CL']== 13]
SJPC_2020 = SJ_2020[SJ_2020['NPA_CL']== 31]
CHPC_2020 = CH_2020[CH_2020['NPA_CL']== 19]

print('대전청 교통사고 관리 건 수 ', len(DJPC_2020))
print('세종청 교통사고 관리 건 수 ', len(SJPC_2020))
print('충남청 교통사고 관리 건 수 ', len(CHPC_2020))

대전청 교통사고 관리 건 수  25896
세종청 교통사고 관리 건 수  262
충남청 교통사고 관리 건 수  40802


### 2021년 데이터

In [496]:
len(k_2021)

164582

In [497]:
k_2021 = duplicated_location(k_2021)

중복 처리 전 교통사고 건수:  164582
중복 처리 후 교통사고 건수 :  124871


In [498]:
k_2021 = k_2021.fillna({'HPPN_X':k_2021['HPPN_X'].mean()})
k_2021 = k_2021.fillna({'HPPN_Y':k_2021['HPPN_Y'].mean()})
k_2021.isnull().sum()

NPA_CL         0
EVT_STAT_CD    0
EVT_CL_CD      0
HPPN_X         0
HPPN_Y         0
dtype: int64

#### 2021년 충남 교통사고

In [499]:
# 충남 - 대전 
CH_2021 = pd.DataFrame()
CH_2021 = translocation_CH(CH_2021, k_2021, 127.55529876331815, 127.25007536223731, 36.186117749974656, 36.49779017085958 )
print('2021년 충남지역에서 대전지역에 해당하는 데이터 제거 후 개수 : ', len(CH_2021))

# 충남 - 세종   127.23, 127.1, 36.24, 36.43)
CH_2021 = translocation_CH(CH_2021, CH_2021, 127.40746822278025, 127.12869800443497, 36.41194803285692, 36.70492)
print('2021년 충남지역에서 세종지역에 해당하는 데이터 제거 후 개수 : ', len(CH_2021))

CH_2021 = translocation(CH_2021,CH_2021, 127.62692947603428,126.13148910811715,35.98803519542761,37.056951491204124 )
print('2021년 충남지역에서 교통사고 건수 : ', len(CH_2021))

2021년 충남지역에서 대전지역에 해당하는 데이터 제거 후 개수 :  75595
2021년 충남지역에서 세종지역에 해당하는 데이터 제거 후 개수 :  69007
2021년 충남지역에서 교통사고 건수 :  68511


#### 2021년 대전 교통사고

In [500]:
DJ_2021 = pd.DataFrame()
DJ_2021 = translocation(DJ_2021, k_2021, 127.55529876331815, 127.25007536223731, 36.186117749974656, 36.49779017085958)
print('2021년 대전지역에서 교통사고 건수 : ', len(DJ_2021))

2021년 대전지역에서 교통사고 건수 :  49276


#### 2021년 세종 교통사고

In [501]:
SJ_2021 = pd.DataFrame()
SJ_2021 = translocation(SJ_2021, k_2021, 127.40746822278025, 127.12869800443497, 36.41194803285692, 36.70492)
print('2021년 세종지역에서 교통사고 건수 : ', len(SJ_2021))

2021년 세종지역에서 교통사고 건수 :  10321


### 2021년 충남, 대전, 세종 교통사고 합

In [503]:
TA_2021 = pd.concat([CH_2021, DJ_2021, SJ_2021])
len(TA_2021)

128108

In [504]:
TA_2021 = duplicated_location(TA_2021)

중복 처리 전 교통사고 건수:  128108
중복 처리 후 교통사고 건수 :  124375


### 2021년 각 경찰청 별로 관리한 교통사고 건수

In [505]:
DJPC_2021 = DJ_2021[DJ_2021['NPA_CL']== 13]
SJPC_2021 = SJ_2021[SJ_2021['NPA_CL']== 31]
CHPC_2021 = CH_2021[CH_2021['NPA_CL']== 19]

print('대전청 교통사고 관리 건 수 ', len(DJPC_2021))
print('세종청 교통사고 관리 건 수 ', len(SJPC_2021))
print('충남청 교통사고 관리 건 수 ', len(CHPC_2021))

대전청 교통사고 관리 건 수  45776
세종청 교통사고 관리 건 수  7090
충남청 교통사고 관리 건 수  68095


### EDA

##### 13 = 대전청
##### 19 = 충남청
##### 31 = 세종

In [511]:
sample = CDS.head(10)

In [512]:
cctv = CCTV

import folium as f

Daejeon = f.Map(location=[36.5847,126.8999],
    tiles='OpenStreetMap',
    zoom_start=10
)
Daejeon

# 관할 경찰청과 사건상태 변환 함수
name = ['없음', '접수중', '접수', '초기지령', '지령', '도착', '네비종결', '종결요청', '지령취소', '타청이첩', '종결요청']

def TransName(sample):
    for i in range(len(sample)):
        if(sample.iloc[i]['EVT_STAT_CD']==1):
            return name[1]
        elif (sample.iloc[i]['EVT_STAT_CD']==2):
            return name[2]
        elif (sample.iloc[i]['EVT_STAT_CD']==3):
            return name[3]
        elif (sample.iloc[i]['EVT_STAT_CD']==4):
            return name[4]
        elif (sample.iloc[i]['EVT_STAT_CD']==5):
            return name[5]
        elif (sample.iloc[i]['EVT_STAT_CD']==6):
            return name[6]
        elif (sample.iloc[i]['EVT_STAT_CD']==7):
            return name[7]
        elif (sample.iloc[i]['EVT_STAT_CD']==8):
            return name[8]
        elif (sample.iloc[i]['EVT_STAT_CD']==9):
            return name[9]
        elif (sample.iloc[i]['EVT_STAT_CD']==10):
            return name[10]
        else :
            print('해당없음')

def TransPolice(sample):
    for i in range(len(sample)):
        if(sample.iloc[i]['NPA_CL']==1):
            return '본청'
        elif (sample.iloc[i]['NPA_CL']==8):
            return '서울청'
        elif (sample.iloc[i]['NPA_CL']==9):
            return '부산청'
        elif (sample.iloc[i]['NPA_CL']==10):
            return '대구청'
        elif (sample.iloc[i]['NPA_CL']==11):
            return '인천청'
        elif (sample.iloc[i]['NPA_CL']==12):
            return '광주청'
        elif (sample.iloc[i]['NPA_CL']==13):
            return '대전청'
        elif (sample.iloc[i]['NPA_CL']==14):
            return '울산청'
        elif (sample.iloc[i]['NPA_CL']==15):
            return '경기남부청'
        elif (sample.iloc[i]['NPA_CL']==16):
            return '경기북부청'
        elif (sample.iloc[i]['NPA_CL']==17):
            return '강원청'
        elif (sample.iloc[i]['NPA_CL']==18):
            return '충북청'
        elif (sample.iloc[i]['NPA_CL']==19):
            return '충남청'
        elif (sample.iloc[i]['NPA_CL']==20):
            return '전북청'
        elif (sample.iloc[i]['NPA_CL']==21):
            return '전남청'
        elif (sample.iloc[i]['NPA_CL']==22):
            return '경북청'
        elif (sample.iloc[i]['NPA_CL']==23):
            return '경남청'
        elif (sample.iloc[i]['NPA_CL']==24):
            return '제주청'
        elif (sample.iloc[i]['NPA_CL']==31):
            return '세종청'
        else :
            print('해당없음')

for i in range(len(sample)):
    marker_Daejeon = f.Marker([sample.iloc[i]['HPPN_Y'],sample.iloc[i]['HPPN_X']],
    #popup=sample.iloc[i]['EVT_STAT_CD'],
    popup = "{0}, {1}".format(TransName(sample),TransPolice(sample)),
    icon = f.Icon(color='blue',icon='star'))

    marker_Daejeon.add_to(Daejeon)
Daejeon

https://m.blog.naver.com/rackhunson/222401975250

## 충남, 대전, 세종 교통사고 발생 위치 (로딩 완전 김)

In [516]:
from folium import Marker
from folium.plugins import MarkerCluster



# Draw a basemap
m = f.Map(location=[36.65881035745041,126.67278633997991], tiles='openstreetmap', zoom_start=11)

# Add  points to the map
mc = MarkerCluster()
for _, row in CDS.iterrows():
    mc.add_child(    
        Marker(location = [row['HPPN_Y'], row['HPPN_X']],
        popup= "{0}, {1}".format(TransName(CDS),TransPolice(CDS)),
    )
    )
    
m.add_child(mc)
m.add_child(f.LatLngPopup()) # 클릭시 위도, 경도 출력

# Display the map
m

## 지역별 교통사고 발생위치

### 충남지역 교통사고

m = f.Map(location=[36.5847,126.8999], tiles='openstreetmap', zoom_start=11)

# Add  points to the map
mc = MarkerCluster()
for _, row in Chungnam.iterrows():
    mc.add_child(    
        Marker(location = [row['HPPN_Y'], row['HPPN_X']],
        popup= "{0}, {1}".format(TransName(Chungnam),TransPolice(Chungnam)),
        )
    )
    
m.add_child(mc)
m.add_child(f.LatLngPopup()) # 클릭시 위도, 경도 출력

# Display the map
m

### 대전지역 교통사고

m = f.Map(location=[36.3504119,127.3845475], tiles='openstreetmap', zoom_start=11)

# Add  points to the map
mc = MarkerCluster()
for _, row in DAEJEON.iterrows():
    mc.add_child(    
        Marker(location = [row['HPPN_Y'], row['HPPN_X']],
        popup= "{0}, {1}".format(TransName(DAEJEON),TransPolice(DAEJEON)),
        )
    )
    
m.add_child(mc)
m.add_child(f.LatLngPopup()) # 클릭시 위도, 경도 출력

# Display the map
m

### 세종지역 교통사고

m = f.Map(location=[36.48,127.29], tiles='openstreetmap', zoom_start=11)

# Add  points to the map
mc = MarkerCluster()
for _, row in SEJONG.iterrows():
    mc.add_child(    
        Marker(location = [row['HPPN_Y'], row['HPPN_X']],
        popup= "{0}, {1}".format(TransName(SEJONG),TransPolice(SEJONG)),
        )
    )
    
m.add_child(mc)
m.add_child(f.LatLngPopup()) # 클릭시 위도, 경도 출력

# Display the map
m

## 연도별 충남, 대전, 세종 교통사고 발생 위치

### 2020년 충남 교통사고 발생 위치

m = f.Map(location=[36.5847,126.8999], tiles='openstreetmap', zoom_start=11)

# Add  points to the map
mc = MarkerCluster()
for _, row in CH_2020.iterrows():
    mc.add_child(    
        Marker(location = [row['HPPN_Y'], row['HPPN_X']],
        popup= "{0}, {1}".format(TransName(CH_2020),TransPolice(CH_2020)),
        )
    )
    
m.add_child(mc)
m.add_child(f.LatLngPopup()) # 클릭시 위도, 경도 출력

# Display the map
m

### 2020년 대전 교통사고 발생 위치

m = f.Map(location=[36.3504119,127.3845475], tiles='openstreetmap', zoom_start=11)

# Add  points to the map
mc = MarkerCluster()
for _, row in DJ_2020.iterrows():
    mc.add_child(    
        Marker(location = [row['HPPN_Y'], row['HPPN_X']],
        popup= "{0}, {1}".format(TransName(DJ_2020),TransPolice(DJ_2020)),
        )
    )
    
m.add_child(mc)
m.add_child(f.LatLngPopup()) # 클릭시 위도, 경도 출력

# Display the map
m

### 2020년 세종 교통사고 발생 위치

m = f.Map(location=[36.48,127.29], tiles='openstreetmap', zoom_start=11)

# Add  points to the map
mc = MarkerCluster()
for _, row in SJ_2020.iterrows():
    mc.add_child(    
        Marker(location = [row['HPPN_Y'], row['HPPN_X']],
        popup= "{0}, {1}".format(TransName(SJ_2020),TransPolice(SJ_2020)),
        )
    )
    
m.add_child(mc)
m.add_child(f.LatLngPopup()) # 클릭시 위도, 경도 출력

# Display the map
m

### 2021년 충남 교통사고 발생 위치

m = f.Map(location=[36.5847,126.8999], tiles='openstreetmap', zoom_start=11)

# Add  points to the map
mc = MarkerCluster()
for _, row in CH_2021.iterrows():
    mc.add_child(    
        Marker(location = [row['HPPN_Y'], row['HPPN_X']],
        popup= "{0}, {1}".format(TransName(CH_2021),TransPolice(CH_2021)),
        )
    )
    
m.add_child(mc)
m.add_child(f.LatLngPopup()) # 클릭시 위도, 경도 출력

# Display the map
m

### 2021년 대전 교통사고 발생 위치

m = f.Map(location=[36.3504119,127.3845475], tiles='openstreetmap', zoom_start=11)

# Add  points to the map
mc = MarkerCluster()
for _, row in DJ_2021.iterrows():
    mc.add_child(    
        Marker(location = [row['HPPN_Y'], row['HPPN_X']],
        popup= "{0}, {1}".format(TransName(DJ_2021),TransPolice(DJ_2021)),
        )
    )
    
m.add_child(mc)
m.add_child(f.LatLngPopup()) # 클릭시 위도, 경도 출력

# Display the map
m

### 2021년 세종 교통사고 발생 위치

m = f.Map(location=[36.48,127.29], tiles='openstreetmap', zoom_start=11)

# Add  points to the map
mc = MarkerCluster()
for _, row in SJ_2021.iterrows():
    mc.add_child(    
        Marker(location = [row['HPPN_Y'], row['HPPN_X']],
        popup= "{0}, {1}".format(TransName(SJ_2021),TransPolice(SJ_2021)),
        )
    )
    
m.add_child(mc)
m.add_child(f.LatLngPopup()) # 클릭시 위도, 경도 출력

# Display the map
m

## 각 청에서 관리하는 교통사고 사건

### 대전청에서 관리한 교통사고 사건

m = f.Map(location=[36.3504119,127.3845475], tiles='openstreetmap', zoom_start=11)

# Add  points to the map
mc = MarkerCluster()
for _, row in DJPC.iterrows():
    mc.add_child(    
        Marker(location = [row['HPPN_Y'], row['HPPN_X']],
        popup= "{0}, {1}".format(TransName(DJPC),TransPolice(DJPC)),
        )
    )
    
m.add_child(mc)
m.add_child(f.LatLngPopup()) # 클릭시 위도, 경도 출력

# Display the map
m

### 세종청에서 관리한 교통사고 사건

m = f.Map(location=[36.48,127.29], tiles='openstreetmap', zoom_start=11)

# Add  points to the map
mc = MarkerCluster()
for _, row in SJPC.iterrows():
    mc.add_child(    
        Marker(location = [row['HPPN_Y'], row['HPPN_X']],
              popup= "{0}, {1}".format(TransName(SJPC),TransPolice(SJPC)),
              )
    )
    
m.add_child(mc)
m.add_child(f.LatLngPopup()) # 클릭시 위도, 경도 출력

# Display the map
m

### 충남청에서 관리한 교통사고 사건

m = f.Map(location=[36.65881035745041,126.67278633997991], tiles='openstreetmap', zoom_start=11)

# Add  points to the map
mc = MarkerCluster()
for _, row in CHPC.iterrows():
    mc.add_child(    
        Marker(location = [row['HPPN_Y'], row['HPPN_X']],
              popup= "{0}, {1}".format(TransName(CHPC),TransPolice(CHPC)),
              )
    )
    
m.add_child(mc)
m.add_child(f.LatLngPopup()) # 클릭시 위도, 경도 출력



# Display the map
m

## CCTV

### 맵 초기화

cctv_map = f.Map(location=[36.5847,126.8999],
    tiles='OpenStreetMap',
    zoom_start=10
)


### 충남 cctv

cctv_map = f.Map(location=[36.5847,126.8999],
    tiles='OpenStreetMap',
    zoom_start=10
)

for i in range(len(CH_CCTV)):
    marker_cctv = f.Marker([CH_CCTV.iloc[i]['위도'],CH_CCTV.iloc[i]['경도']],
    popup = "{0}, {1}".format(CH_CCTV.iloc[i]['도로노선명'],CH_CCTV.iloc[i]['설치장소']),
    icon = f.Icon(color='red',icon='star'))

    marker_cctv.add_to(cctv_map)
cctv_map

### 대전 cctv

cctv_map = f.Map(location=[36.5847,126.8999],
    tiles='OpenStreetMap',
    zoom_start=10
)


for i in range(len(DJ_CCTV)):
    marker_cctv = f.Marker([DJ_CCTV.iloc[i]['위도'],DJ_CCTV.iloc[i]['경도']],
    #popup=sample.iloc[i]['EVT_STAT_CD'],
    popup = "{0}, {1}".format(DJ_CCTV.iloc[i]['도로노선명'],DJ_CCTV.iloc[i]['설치장소']),
    icon = f.Icon(color='red',icon='star'))

    marker_cctv.add_to(cctv_map)
cctv_map

### 세종 CCTV

cctv_map = f.Map(location=[36.5847,126.8999],
    tiles='OpenStreetMap',
    zoom_start=10
)


for i in range(len(SJ_CCTV)):
    marker_cctv = f.Marker([SJ_CCTV.iloc[i]['위도'],SJ_CCTV.iloc[i]['경도']],
    popup = "{0}, {1}".format(SJ_CCTV.iloc[i]['도로노선명'],SJ_CCTV.iloc[i]['설치장소']),
    icon = f.Icon(color='red',icon='star'))

    marker_cctv.add_to(cctv_map)
cctv_map